### PR TITLE
Document NixOS release process #4442

### PR DESCRIPTION
**Pull request by domenkozar**
Fixes https://github.com/NixOS/nixpkgs/issues/4442

Missing:
- [x] https://github.com/NixOS/nixpkgs/commit/40fd9ae3ac8048758abdcfc7d28a78b5f22fe97e
- [x] https://github.com/NixOS/nixos-org-configurations/pull/18
- [x] set `stableBranch` on Hydra jobsets to true (also on -small)
- [x] write changelogs before the branch off

cc @edolstra

### CODE REVIEW

File: nixos/doc/manual/development/releases.xml [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+        <para>
+          edit changelog at
+          <literal>nixos/doc/manual/release-notes/rl-1509.xml</literal>
+          (double check desktop versions are noted)
+        </para>
+        <itemizedlist spacing="compact">
+          <listitem>
+            <para>
+              get all new NixOS modules
+              <literal>git di release-14.12..release-15.09 nixos/modules/module-list.nix|grep ^+</literal>
```
Comment (globin) [inline nixos/doc/manual/development/releases.xml]: `git di` -> `git diff`

**Comment by domenkozar** [comment]
@globin thanks, fixed.

File: nixos/doc/manual/development/releases.xml [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    <title>One month before the release</title>
+    <itemizedlist spacing="compact">
+      <listitem>
+        <para>
+          send a warning email to nix-dev
+        </para>
+      </listitem>
+      <listitem>
+        <para>
+          talk to Eelco and others what should be part of the release
```
Comment (lancelotsix) [inline nixos/doc/manual/development/releases.xml]: Maybe add more details about how to get something into the release ? (set the milestone in github PRs / issues + the blocker tag for the really important matters that should be addressed before release). The “talk to Eelco and others” part seems hard to track for an external contributor

Comment (domenkozar) [inline nixos/doc/manual/development/releases.xml]: Agreed, this was just my mental note that needs to be articulated to fit into the manual.

File: nixos/doc/manual/development/releases.xml [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+      </listitem>
+    </itemizedlist>
+  </section>
+  <section xml:id="at-branch-off-time">
+    <title>At branch off time</title>
+    <itemizedlist spacing="compact">
+      <listitem>
+        <para>
+          <literal>git tag -a -m &quot;Release 15.09-beta&quot; 15.09-branch-off &amp;&amp; git push --tags</literal>
+          (Note: the the commit that should be tagged is the last in
```
Comment (rasendubi) [inline nixos/doc/manual/development/releases.xml]: the the

**Comment by domenkozar** [comment]
I've pushed changes with typo fixes and things that I've done differently/additionally for 16.09

This is far from perfect, but I think it's a good start compared to nothing.

**Pull request merged**